### PR TITLE
#7960 Add JSON logs parsing into pod-logs-viewer on click

### PR DESCRIPTION
**Pull request by Sujimoshi**
Note on DCO:

If the DCO action in the integration test fails, one or more of your commits are not signed off. Please click on the *Details* link next to the DCO action for instructions on how to resolve this.

Fixes [ISSUE #7960]

Checklist:

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community. Feature request: https://github.com/argoproj/argo-cd/issues/7960
* [x] The title of the PR states what changed and the related issues number (used for the release note).
* [x] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [x] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [x] Does this PR require documentation updates?
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ ] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ ] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

**Comment by pasha-codefresh** [comment]
It would be nice if you can improve this feature and add support also when "Wrap lines" feature is enabled

File: ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx [hunk 1]
```diff
@@ -400,3 +400,12 @@ const logNavigators = (actions: NavActions, darkMode: boolean, info?: PageInfo)
         </div>
     );
 };
+
+const renderLine = (line: string, isSelected: boolean) => {
+    if (!isSelected) return line
+    try { 
+        return JSON.stringify(JSON.parse(line), null, 2)
```
Comment (pasha-codefresh) [inline ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx:407]: Please change code according linter 

Снимок экрана 2022-03-19 в 16 09 26

**Review by pasha-codefresh — CHANGES_REQUESTED** [review]

File: ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx (hunk 1 above)
Comment (Sujimoshi) [reply ui/src/app/applications/components/pod-logs-viewer/pod-logs-viewer.tsx]: Linter fixed

**Comment by Sujimoshi** [comment]
> It would be nice if you can improve this feature and add support also when "Wrap lines" feature is enabled
> 
> Снимок экрана 2022-03-19 в 16 08 11

It should correctly work with `whiteSpace: wrap` wich used for wrap lines

**Comment by ebuildy** [comment]
hey what happen to this PR ?

Also will be great to have pretty colorized JSON

**Comment by presidenten** [comment]
@pasha-codefresh 
Please continue the review so we can get the json logs parsing

**Review by pasha-codefresh — APPROVED** [review]
LGTM, thank you!

**Review by todaywasawesome — APPROVED** [review]
LGTM

**Comment by pasha-codefresh** [comment]
@Sujimoshi , i am sorry for delays in review, could you please resolve conflicts ?

**Comment by kencochrane** [comment]
@Sujimoshi thank you for working on this, if you can help resolve the conflicts we can get it merged. If you no longer have interest in working on this, let us know and we can see if we can find someone else to take it over for you. Thanks again!

**Comment by nitishfy** [comment]
@Sujimoshi do you have the bandwidth to resolve the conflicts? 

Once the conflicts are resolved, we can merge this PR. If you don't have the bandwidth, please let us know so that someone else can take up this taks. Thanks!

**Comment by pasha-codefresh** [comment]
I am closing it for now, because even fix conflicts is pretty hard right now, due to a lot of changes in pod-logs-viewer.ts 

@Sujimoshi please let me know if you want to continue to work on this fix. Thanks in advance

**Pull request closed without merging**